component/src/main/java/org/wso2/extension/siddhi/execution/tensorflow/util/CoreUtils.java

```diff
@@ -0,0 +1,324 @@
+                    for (byte value : byteArray) {
+                        objectList.add((int) value);
+                    }
+                } else {
+                    for (byte value : byteArray) {
+                        if (value == 1) {
+                            objectList.add(true);
+                        } else {
+                            objectList.add(false);
+                        }
+                    }
+                }
+            }
+
+            outputTensor.close();
+        }
+        Object[] outputs = new Object[objectList.size()];
+        objectList.toArray(outputs);
+        return outputs;
+    }
+
+    public static long[] getShapeOfArrayAsString(String arrayAsString) {
+        arrayAsString = arrayAsString.substring(arrayAsString.lastIndexOf(":") + 1);
+        int noDims = 0;
+
+        for (int i = 0; i < arrayAsString.length(); i++) {
+            if (arrayAsString.charAt(i) == ' ') {
+                continue;
+            } else if (arrayAsString.charAt(i) == '[') {
+                noDims++;
+            } else {
+                break;
+            }
+        }
+
+        long[] shape = new long[noDims];
+
+        Stack bracketStack = new Stack();
+        Stack countStack = new Stack();
+
+        for (int i = 0; i < arrayAsString.length(); i++) {
+
+            if (arrayAsString.charAt(i) == '[') {
+                bracketStack.push("[");
+                countStack.push(0);
+
+            } else if (arrayAsString.charAt(i) == ',') {
+                int tempNum = (int) countStack.pop();
+                countStack.push(tempNum + 1);
+
+            } else if (arrayAsString.charAt(i) == ']') {
+                int tempSize = (int) countStack.pop();
+                tempSize = tempSize + 1;
+                bracketStack.pop();
+
+                if (shape[bracketStack.size()] == 0) {
+                    shape[bracketStack.size()] = tempSize;
+
+                } else if (!(shape[bracketStack.size()] == tempSize)) {
+                    throw new SiddhiAppRuntimeException("Array size is inconsistent");
+                }
+            }
+        }
+
+        return shape;
+    }
+
+    public static Tensor createTensor(String arrayAsString) throws CharacterCodingException {
+        long[] shape = getShapeOfArrayAsString(arrayAsString);
+        Tensor tensor;
+
+        String dataType = arrayAsString.substring(0, arrayAsString.lastIndexOf(":"));
+        dataType = dataType.replaceAll("\\s", "");
+        arrayAsString = arrayAsString.substring(arrayAsString.lastIndexOf(":") + 1);
+
+        arrayAsString = arrayAsString.replaceAll("]", " ");
+        arrayAsString = arrayAsString.replaceAll(",", " ");
+        arrayAsString = arrayAsString.replaceAll("\\[", " ");
+        String[] stringArray = arrayAsString.split(" ");
+
+        List tempList = new LinkedList();
+
+        for (String string : stringArray) {
+            if (string != null && string.length() > 0) {
```

Comment:
use isEmpty method rather using length to check whether a string is empty or not.

Reply:
done